File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.kt
class Solution {
fun findGCD(nums: IntArray): Int {
    return gcd(nums.min()!!, nums.max()!!)
}
fun gcd(p: Int, q: Int): Int {
    return if (q == 0) p 
    else gcd(q, p % q)
}
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        ma = max(nums)
        mi = min(nums)
        
        return gcd(ma,mi)
    
    def gcd(self, left, right):
        i = left
        while (True):
            if i%ma ==0 and i% mi == 0:
                return i
            